library/src/main/java/com/felipecsl/gifimageview/library/GifImageView.java

```diff
@@ -122,53 +133,50 @@ public int getGifHeight() {
   }
 
   @Override public void run() {
-    if (shouldClear) {
-      handler.post(cleanupRunnable);
-      return;
-    }
-
-    final int n = gifDecoder.getFrameCount();
     do {
-      for (int i = 0; i < n; i++) {
-        if (!animating) {
-          break;
-        }
-        //milliseconds spent on frame decode
-        long frameDecodeTime = 0;
-        try {
-          long before = System.nanoTime();
-          tmpBitmap = gifDecoder.getNextFrame();
-          frameDecodeTime = (System.nanoTime() - before) / 1000000;
-          if (frameCallback != null) {
-            tmpBitmap = frameCallback.onFrameAvailable(tmpBitmap);
-          }
-
-          if (!animating) {
-            break;
-          }
-          handler.post(updateResults);
-        } catch (final ArrayIndexOutOfBoundsException | IllegalArgumentException e) {
-          Log.w(TAG, e);
-        }
-        if (!animating) {
-          break;
+      if (!animating && !renderFrame) {
+        break;
+      }
+      boolean advance = gifDecoder.advance();
+
+      //milliseconds spent on frame decode
+      long frameDecodeTime = 0;
+      try {
+        long before = System.nanoTime();
+        tmpBitmap = gifDecoder.getNextFrame();
+        if (frameCallback != null) {
+          tmpBitmap = frameCallback.onFrameAvailable(tmpBitmap);
         }
-        gifDecoder.advance();
-        try {
-          int delay = gifDecoder.getNextDelay();
-          // Sleep for frame duration minus time already spent on frame decode
-          // Actually we need next frame decode duration here,
-          // but I use previous frame time to make code more readable
-          delay -= frameDecodeTime;
-          if (delay > 0) {
-            Thread.sleep(framesDisplayDuration > 0 ? framesDisplayDuration : delay);
-          }
-        } catch (final Exception e) {
-          // suppress any exception
-          // it can be InterruptedException or IllegalArgumentException
+        frameDecodeTime = (System.nanoTime() - before) / 1000000;
+        handler.post(updateResults);
+      } catch (final ArrayIndexOutOfBoundsException | IllegalArgumentException e) {
+        Log.w(TAG, e);
+      }
+
+      renderFrame = false;
+      if (!animating || !advance) {
```

Comment:
It's not about saving 1 character, it's about saving a operation. Anw, it's coding style, it's up to individuals.